src/Microsoft.Health.Dicom.Tests.Common/Comparers/BinaryComparer.cs

```diff
@@ -0,0 +1,453 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Hashing;
+
+namespace Microsoft.Health.Dicom.Tests.Common.Comparers;
+
+#nullable enable
+
+/// <summary>
+/// Represents both an <see cref="IComparer{T}"/> and <see cref="IEqualityComparer{T}"/> for contiguous regions of memory.
+/// </summary>
+/// <remarks>
+/// <see cref="ReadOnlySpan{T}"/> is a ref struct and cannot be used as a type argument for
+/// <see cref="IComparer{T}"/> and <see cref="IEqualityComparer{T}"/>. However, equivalent methods
+/// are available without the interface implementation such as <see cref="Equals(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>.
+/// </remarks>
+public sealed class BinaryComparer : IComparer<byte[]>, IComparer<Stream>, IEqualityComparer<byte[]>, IEqualityComparer<Stream>
```

Comment:
This class is from my own personal development, and I thought it would be a good addition now.